fds/services/dvc_service.py

```diff
@@ -65,85 +74,103 @@ def __skip_already_added(self, root, dirs) -> None:
+            if answers["selection_choice"] == DvcChoices.ADD_TO_DVC.value:
                 # Dont need to traverse deep
                 [dirs.remove(d) for d in list(dirs)]
                 return file_or_dir_to_check
-            elif answers["selection_choice"] == "Add to Git":
+            elif answers["selection_choice"] == DvcChoices.ADD_TO_GIT.value:
                 # Dont need to traverse deep
                 [dirs.remove(d) for d in list(dirs)]
                 return
-            elif answers["selection_choice"] == "Ignore":
+            elif answers["selection_choice"] == DvcChoices.IGNORE.value:
                 # We should ignore the ./ in beginning when adding to gitignore
                 # Add files to gitignore
                 append_line_to_file(".gitignore", file_or_dir_to_check[file_or_dir_to_check.startswith('./') and 2:])
                 # Dont need to traverse deep
                 [dirs.remove(d) for d in list(dirs)]
                 return
+        return
 
     def __add(self, add_argument: str):
         chosen_files_or_folders = []
+        # Keep track of dirs which are below threshold size, so we dont iterate the files inside these dirs
+        skipped_dirs = []
         # May be add all the folders given in the .gitignore
         folders_to_exclude = ['.git', '.dvc']
         if add_argument == AddCommands.ALL.value:
             path_to_walk = self.repo_path
         else:
             path_to_walk = f"{self.repo_path}/{add_argument}"
+        # if argument is to add a file
+        if os.path.isfile(path_to_walk) and get_size_of_path(path_to_walk) >= MAX_THRESHOLD_SIZE:
+            # Keep the file in chosen list
+            chosen_files_or_folders = [path_to_walk]
         for (root, dirs, files) in os.walk(path_to_walk, topdown=True, followlinks=False):
             # Now skip the un-necessary folders
             [dirs.remove(d) for d in list(dirs) if d in folders_to_exclude]
             # Skip the already added files/folders
             self.__skip_already_added(root, dirs)
             # First check root
-            folder_to_add = self.__get_to_add_to_dvc(root, dirs, "Dir")
+            folder_to_add = self.__get_to_add_to_dvc(root, dirs, "Dir", skipped_dirs)
             if folder_to_add is not None:
                 chosen_files_or_folders.append(folder_to_add)
             else:
                 # Only if they dont select the directory then ask for files, otherwise ignore asking about files of the directory
                 # We are also showing if the user chooses to skip becuase the user might not know there is a large file
                 # in the directory and choose skip because he dont want the entire directory to be added.
+                # If the root is skipped because it is below threshold size then we don't need to check files
+                if root in skipped_dirs:
```

Comment:
This is where we use the skipped_dirs, please read the comment above on why we should use it